tests/pytests/integration/modules/test_mac_user.py

```diff
@@ -0,0 +1,262 @@
+"""
+    :codeauthor: Nicole Thomas <nicole@saltstack.com>
+"""
+
+import os
+
+import pytest
+from saltfactories.utils import random_string
+
+import salt.utils.files
+from salt.exceptions import CommandExecutionError
+
+# Create user strings for tests
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.destructive_test,
+    pytest.mark.skip_if_not_root,
+    pytest.mark.skip_unless_on_darwin,
+]
+
+
+@pytest.fixture(scope="function")
+def setup_teardown_vars(salt_call_cli):
+    ret = salt_call_cli.run("grains.item", "kernel")
+    os_grain = ret.data
+    if os_grain["kernel"] not in "Darwin":
+        pytest.skip("Test not applicable to '{kernel}' kernel".format(**os_grain))
+
+    ADD_USER = random_string("RS-", lowercase=False)
+    DEL_USER = random_string("RS-", lowercase=False)
+    PRIMARY_GROUP_USER = random_string("RS-", lowercase=False)
+    CHANGE_USER = random_string("RS-", lowercase=False)
+
+    try:
+        yield ADD_USER, DEL_USER, PRIMARY_GROUP_USER, CHANGE_USER
+    finally:
+        # Delete ADD_USER
+        ret = salt_call_cli.run("user.info", ADD_USER)
+        add_info = ret.data
+        if add_info:
+            salt_call_cli.run("user.delete", ADD_USER)
+
+        # Delete DEL_USER if something failed
+        ret = salt_call_cli.run("user.info", DEL_USER)
+        del_info = ret.data
+        if del_info:
+            salt_call_cli.run("user.delete", DEL_USER)
+
+        # Delete CHANGE_USER
+        ret = salt_call_cli.run("user.info", CHANGE_USER)
+        change_info = ret.data
+        if change_info:
+            salt_call_cli.run("user.delete", CHANGE_USER)
+
+
+def test_mac_user_add(salt_call_cli, setup_teardown_vars):
+    """
+    Tests the add function
+    """
+    ADD_USER = setup_teardown_vars[0]
+
+    try:
+        salt_call_cli.run("user.add", ADD_USER)
+        ret = salt_call_cli.run("user.info", ADD_USER)
+        user_info = ret.data
+        assert ADD_USER == user_info["name"]
+    except CommandExecutionError:
+        salt_call_cli.run("user.delete", ADD_USER)
+        raise
```

Comment:
```suggestion
def test_mac_user_add(salt_call_cli, setup_teardown_vars):
    """
    Tests the add function
    """
    ADD_USER = setup_teardown_vars[0]

    salt_call_cli.run("user.add", ADD_USER)
    ret = salt_call_cli.run("user.info", ADD_USER)
    user_info = ret.data
    assert ADD_USER == user_info["name"]
```
The same for the reamining tests